egret/parsers/rts_gmlc/parser.py

```diff
@@ -0,0 +1,838 @@
+
+    elements = model_data['elements']
+    params_of_interest = {
+        'Generator': { 'PMin MW', 'PMax MW'},
+        'Reserve':   {'Requirement'},
+        'Area':      {'MW Load'}
+        }
+
+    # Add a column to timeseries DF to reference the parsed data 
+    # instead of the file name
+    timeseries_pointer_df['Series'] = None
+
+    # Store the timeseries data in the timeseries DF
+    for idx,row in timeseries_pointer_df.iterrows():
+        # Skip rows we don't ingest
+        if not row['Category'] in params_of_interest:
+            continue
+        if not row['Parameter'] in params_of_interest[row['Category']]:
+            continue
+
+        # Skip generators not in skeleton
+        if row['Category'] == 'Generator' and not row['Object'] in elements['generator']:
+            continue
+        # Skip areas not in skeleton
+        if row['Category'] == 'Area' and not row['Object'] in elements['area']:
+            continue
+
+        is_reserve = (row['Category'] == 'Reserve')
+        if is_reserve:
+            # Skip unrecognized reserve names
+            name = str(row['Object'])
+            if not is_valid_reserve_name(name, model_data):
+                continue
+
+        # Read the timeseries file if we haven't already, using the
+        # canonical file path as a key into previously read filenames.
+        fname = os.path.abspath(os.path.join(rts_gmlc_dir, row['Data File']))
+        if not fname in timeseries_file_map:
+            sim = row['Simulation']
+            data = _read_timeseries_file(fname, minutes_per_period[sim], 
+                                         start_time, end_time, row['Object'])
+            timeseries_file_map[fname] = data
+
+        # Save a reference to the relevant data as a Series
+        timeseries_pointer_df.at[idx,'Series'] = timeseries_file_map[fname][row['Object']]
+
+    # Remove columns that we don't want to preserve
+    keepers= {'Simulation', 'Category', 'Object', 'Parameter', 'Series'}
+    for c in timeseries_pointer_df.columns:
+        if not c in keepers:
+            timeseries_pointer_df.pop(c)
+
+    # Remove irrelevant rows
+    timeseries_pointer_df.dropna(subset=['Series'], inplace=True)
+
+    # Sort by simulation
+    timeseries_pointer_df.sort_values(by='Simulation', inplace=True)
+
+    return timeseries_pointer_df
+
+def _parse_datetimes_if_strings(begin_time:Union[datetime,str], end_time:Union[datetime,str]):
+    '''
+    Ensure both dates are datetimes, parsing date strings if necessary.
+
+    Returns
+    -------
+    begin_time:datetime
+        The begin_time as a datetime, parsing it if necessary
+    end_time:datetime
+        The end_time as a datetime, parsing it if necessary
+    '''
+
+    datetime_format = "%Y-%m-%d %H:%M:%S"
+
+    datestr = "YYYY-DD-MM"
+    midnight = " 00:00:00"
+
+    if isinstance(begin_time,datetime):
+        pass
+    elif isinstance(begin_time,str):
+        if len(begin_time) == len(datestr):

```

Comment:
We should probably throw an error if this is false.

Reply:
We accept time strings both with and without the time of day, defaulting to midnight if the time was left off.  This check determines which of the two accepted formats we have.  If the string is truly not in the right format then strptime will raise a ValueError exception a couple of lines below this. Unless you feel strongly otherwise I plan to leave the code as is.

Reply:
I ended up switching to dateutil.parser.parse(), which infers the date and time format.  See new function _convert_to_datetime().